src/algebra/order/ring.lean

```diff
@@ -366,6 +366,21 @@ by classical; exact decidable.lt_mul_of_one_lt_left
 lemma lt_two_mul_self [nontrivial α] (ha : 0 < a) : a < 2 * a :=
 lt_mul_of_one_lt_left ha one_lt_two
 
+lemma lt_mul_left {n m : α} (hn : 0 < n) (hm : 1 < m) : n < m * n :=
+begin
+  rw [←one_mul n, ←one_mul m, mul_assoc, one_mul (m * _), ←mul_assoc, mul_one],
+  exact mul_lt_mul_of_pos_right hm hn
+end
+
+lemma lt_mul_right {n m : α} (hn : 0 < n) (hm : 1 < m) : n < n * m :=
+begin
+  rw [←mul_one n, mul_assoc, one_mul],
+  exact mul_lt_mul_of_pos_left hm hn
+end
```

Comment:
```suggestion
lemma lt_mul_left {n m : α} (hn : 0 < n) (hm : 1 < m) : n < m * n :=
by { convert mul_lt_mul_of_pos_right hm hn, rw one_mul }

lemma lt_mul_right {n m : α} (hn : 0 < n) (hm : 1 < m) : n < n * m :=
by { convert mul_lt_mul_of_pos_left hm hn, rw mul_one }
```
This seems to be more efficient.

Reply:
(Cannot guarantee the style is correct.)

Reply:
```
import algebra.order.ring

variables {α : Type*} [ordered_semiring α] {a b c d : α}

set_option profiler true

/-
parsing took 5.89ms
-/

/-
type checking of lt_mul_left' took 0.128ms
-/

/-
decl post-processing of lt_mul_left' took 0.0042ms
-/

/-
elaboration of lt_mul_left' took 24.2ms
-/

lemma lt_mul_left' (hn : 0 < a) (hm : 1 < b) : a < b * a :=
begin
  rw [←one_mul a, ←one_mul b, mul_assoc, one_mul (b * _), ←mul_assoc, mul_one],
  exact mul_lt_mul_of_pos_right hm hn
  /-
  elaboration: tactic compilation took 5.54ms
  -/

end

/-
compilation of lt_mul_left' took 0.0018ms
-/

--

/-
parsing took 12.8ms
-/

/-
type checking of lt_mul_left'' took 0.106ms
-/

/-
decl post-processing of lt_mul_left'' took 0.0033ms
-/

/-
elaboration of lt_mul_left'' took 33.4ms
-/

lemma lt_mul_left'' (hn : 0 < a) (hm : 1 < b) : a < b * a :=
/-
elaboration: tactic compilation took 6.65ms
-/

by { convert mul_lt_mul_of_pos_right hm hn, rw one_mul }
```
I'll switch to your method because the proof is "clearer", but I do want to point out that "efficient" is a poorly defined term, and for some definitions of it, the `rw`-based proof is more "efficient". For one, the `rw` based term acts almost entirely on a syntax level.